### ISSUE TITLE
Freeze specific layers for fine tuning

### ISSUE THREAD

**Issue by pantelmen** [Deci-AI/super-gradients#1089]
### 🚀 Feature Request

Hello, i need to freeze specific layers (backbone) and train other layers for fine tuning. It is possible now? or it is a feature that we can wait for

**Comment by harpreetsahota204** [comment]
Hi @pantelmen. YOLO-NAS is a PyTorch model, so you can freeze layers as your normally would by setting `requires_grad=False` for the layer you wish to freeze. 

Have you tried this already? If you are facing errors, please share a reproducible notebook so we can investigate further. 

Cheers

**Comment by BloodAxe** [comment]
For freezing certain layers you can change `requires_grad` of corresponding layers as follows:

```
from super_gradients.common.object_names import Models
from super_gradients.training import models


def main():
    model = models.get(Models.YOLO_NAS_S, pretrained_weights="coco")

    trainable_parameters = sum([p.numel() for p in model.parameters() if p.requires_grad])

    for m in model.backbone.parameters():
        m.requires_grad = False

    trainable_parameters_after = sum([p.numel() for p in model.parameters() if p.requires_grad])

    print(trainable_parameters)
    print(trainable_parameters_after)


if __name__ == "__main__":
    main()
```

The code above should print:
```
Total parameters 19053888
After freezing backbone 4209380
```
